File: client/react/hrms/src/services/employerStatusService.js
import axios from "axios";

export default class EmployerStatusService {
  add(employerStatus) {
    return axios.post(
      "http://localhost:8080/api/employerstatuses/add",
      employerStatus
    );
  }

  getAll() {
    return axios.get("http://localhost:8080/api/employerstatuses/getall");
  }

  getLastByEmployerId(employerId) {
    return axios.get(
      "http://localhost:8080/api/employerstatuses/getlastbyemployerid?employerId=" +
        employerId
    );
  }
}

File: server/java/src/main/java/com/hrms/business/abstracts/EmployerStatusService.java
package com.hrms.business.abstracts;

import com.hrms.core.utilities.results.DataResult;
import com.hrms.core.utilities.results.Result;
import com.hrms.entities.concretes.EmployerStatus;
import com.hrms.entities.concretes.JobPostingStatus;

import java.util.List;

public interface EmployerStatusService {
    Result add(EmployerStatus employerStatus);
    DataResult<EmployerStatus> getLastByEmployerId(int employerId);
    DataResult<List<EmployerStatus>> getAllByEmployerId(int employerId);
    DataResult<List<EmployerStatus>> getAll();
}
